openssl/src/ssl/mod.rs

```diff
@@ -3114,6 +3112,142 @@ impl SslRef {
         }
         Ok(())
     }
+    #[cfg(not(boringssl))]
+    pub fn set_method(&mut self, method: SslMethod) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::SSL_set_ssl_method(self.as_ptr(), method.as_ptr()))?;
+        };
+        Ok(())
+    }
+    #[corresponds(SSL_use_Private_Key_file)]
+    pub fn set_private_key_file<P: AsRef<Path>>(
+        &mut self,
+        path: P,
+        ssl_file_type: SslFiletype,
+    ) -> Result<(), ErrorStack> {
+        let p = path.as_ref().as_os_str().to_str().unwrap();
+        let key_file = CString::new(p).unwrap();
+        unsafe {
+            cvt(ffi::SSL_use_PrivateKey_file(
+                self.as_ptr(),
+                key_file.as_ptr(),
+                ssl_file_type.as_raw(),
+            ))?;
+        };
+        Ok(())
+    }
+    #[corresponds(SSL_use_PrivateKey)]
+    pub fn use_private_key_pem(&mut self, key: &[u8]) -> Result<(), ErrorStack> {
+        let pkey = PKey::private_key_from_pem(key)?;
+        unsafe {
+            // take ownership of pkey
+            cvt(ffi::SSL_use_PrivateKey(self.as_ptr(), pkey.as_ptr()))?;
+        };
+        mem::forget(pkey);
+        Ok(())
+    }
+    #[corresponds(SSL_use_certificate)]
+    pub fn use_certificate_pem(&mut self, cert: &[u8]) -> Result<(), ErrorStack> {
```

Comment:
Similarly, this function should take a `&X509Ref` and be called `set_certificate`